### PR TITLE
Photovoltaics Changes

### LINKED ISSUE #53: Add new characteristics for PV (and wind if added)
Add "Year Installed, "Annual Output", "Levelized Cost of Electricity"

### LINKED ISSUE #54: PV financing
Add whether the panels are leased, self-financed, PPA or other.

### PR DESCRIPTION
**Pull request by nmerket**
- [x] Need definition for `AnnualOutput`.
- [x] Need definition for `LevelizedCostOfElectricity`.
- [x] Need another set of eyes on PV ownership enumerations.

baseelements_pvsystem

@RebHudson
Fixes #53 
Fixed #54

### CODE REVIEW

File: schemas/HPXMLDataTypes.xsd [hunk 1]
```diff
@@ -1732,6 +1732,9 @@
 		<xs:restriction base="xs:string">
 			<xs:enumeration value="leased"/>
 			<xs:enumeration value="owned"/>
+			<xs:enumeration value="power purchase agreement"/>
+			<xs:enumeration value="utility owned"/>
+			<xs:enumeration value="other"/>
```
Comment (nmerket) [inline schemas/HPXMLDataTypes.xsd]: @Gavinhastings @RebHudson Do these enumerations for `PVSystem/Ownership` looks reasonable?

**Comment by nmerket** [comment]
From @RebHudson
<hr/>
Annual Output could be “actual” or “projected”. These would need to be enumerations for the value. Actual would be a measured value of the full year’s energy output in kWh/year. It would be great to have a year associated with that value.
The projected value requires a calculation/estimation. Does this need to be a set calculation? I could see that different solar installers might use different formulas.
 
There is a simplified version of an output calculator:
http://pvwatts.nrel.gov/
 
For levelized cost of electricity, it also involves a calculation that could be done in different ways. A good definition from SAM (NREL created a SAM model that has a method for calculating LCOE https://www.nrel.gov/analysis/sam/help/html-php/index.html?mtf_lcoe.htm) is: The LCOE is the total cost of installing and operating a project expressed in dollars per kilowatt-hour of electricity generated by the system over its life.
Again – does this need a specific formula associated with it? Or can it be up to the contractor’s discretion?
 
http://www.nrel.gov/analysis/tech_lcoe.html seems to have a simpler version.

File: schemas/HPXMLDataTypes.xsd (hunk 1 above)
Comment (RebHudson) [inline schemas/HPXMLDataTypes.xsd]: Those enumerations work for me

**Comment by nmerket** [comment]
I added the following definitions

### AnnualOutput

> [kWh] Projected Annual Output for a typical meterological year as determined by PVWatts. 

I decided to go with the projected because that's what more people are likely to have. It's fairly simple to calculate too.

### Levelized Cost of Electricity

> [$] The LCOE is the total cost of installing and operating a project expressed in dollars per kilowatt-hour of electricity generated by the system over its life. Can be calculated with System Advisor Model or through a simplified calculation at http://www.nrel.gov/analysis/tech_lcoe.html.

File: schemas/BaseElements.xsd [hunk 2]
```diff
@@ -1457,6 +1457,19 @@
...
+									<xs:element minOccurs="0" name="AnnualOutput"
+										type="RatedAnnualkWh">
+										<xs:annotation>
+											<xs:documentation>[kWh] Projected Annual Output for a typical meterological year as determined by PVWatts. </xs:documentation>
+										</xs:annotation>
+									</xs:element>
+									<xs:element minOccurs="0" name="LevelizedCostOfElectricity"
+										type="Cost">
+										<xs:annotation>
+											<xs:documentation>[$] The LCOE is the total cost of installing and operating a project expressed in dollars per kilowatt-hour of electricity generated by the system over its life. Can be calculated with System Advisor Model or through a simplified calculation at http://www.nrel.gov/analysis/tech_lcoe.html.</xs:documentation>
```
Comment (nmerket) [inline schemas/BaseElements.xsd]: Add "or similar" to doc string.

File: schemas/BaseElements.xsd [hunk 3]
```diff
@@ -1457,6 +1457,19 @@
 										type="Year"/>
 									<xs:element minOccurs="0" name="YearModulesManufactured"
 										type="Year"/>
+									<xs:element minOccurs="0" name="YearInstalled" type="Year"/>
+									<xs:element minOccurs="0" name="AnnualOutput"
+										type="RatedAnnualkWh">
+										<xs:annotation>
+											<xs:documentation>[kWh] Projected Annual Output for a typical meterological year as determined by PVWatts. </xs:documentation>
```
Comment (nmerket) [inline schemas/BaseElements.xsd]: add "or similar"

File: schemas/BaseElements.xsd (hunk 2 above)
Comment (nmerket) [inline schemas/BaseElements.xsd]: add "or similar"

**Comment by nmerket** [comment]
Review: @brandongallagher @GamalielL

**Comment by GamalielL** [comment]
Looks good to me.  I've asked Andy to review as well.

**Comment by brandongallagher** [comment]
works for me.  i'm looking forward to the AnnualOutput field.  that will help us.

**Pull request merged**